### PR TITLE
NOISSUE - Add optional Filename

### PR DESCRIPTION
**Pull request by SammyOina**
# What type of PR is this?
This is a feature as it adds filename to datasets which are optionally verified if specified on the manifest. Filename is also persisted on dataset directory

## What does this do?
- **New Features**
	- Introduced a new field for dataset filenames, enhancing the context and utility of datasets.
	- Added a new error for filename mismatches to improve error handling during dataset validation.

- **Bug Fixes**
	- Enhanced error messaging and handling for better user feedback and debugging.

- **Chores**
	- Updated mock code generation tool version for consistency and maintenance.

## Have you included tests for your changes?
No

## Did you document any new/modified feature?
No

### Notes
```mermaid
sequenceDiagram
    participant User
    participant ManagerService
    participant AgentService
    participant Dataset


    User->>ManagerService: Initiate computation run
    ManagerService->>AgentService: Prepare manifest containing datasets with Filename
    AgentService->>Dataset: Append new dataset with Filename to manifest
    User->>AgentService:Upload dataset
    AgentService->>Dataset:Verify filename as in manifest
    Dataset->>AgentService:Dataset stored with its filename
    AgentService-->>User: Datasets uploaded successfully
```

### CODE REVIEW

**Review by WashingtonKK — CHANGES_REQUESTED** [review]
I am still able to upload a file with a different file name successfuily, without any error.

```
➜  cocos git:(filename) ✗ ./build/cocos-cli data test/manual/data/iris.csv private.pem                                                 
2024/08/08 17:27:25 Uploading dataset CSV: test/manual/data/iris.csv
Uploading data...  100% [============================================================================================================>] 
2024/08/08 17:27:25 Successfully uploaded dataset
```

File: agent/service.go [hunk 1]
```diff
@@ -185,9 +187,14 @@ func (as *agentService) Data(ctx context.Context, dataset Dataset) error {
...
 	}
+
+	if as.computation.Datasets[index].Filename != "" && as.computation.Datasets[index].Filename != dataset.Filename {
+		return ErrFileNameMismatch
+	}
+
 	as.computation.Datasets = slices.Delete(as.computation.Datasets, index, index+1)
 
-	f, err := os.Create(fmt.Sprintf("%s/dataset-%d", algorithm.DatasetsDir, index))
+	f, err := os.Create(fmt.Sprintf("%s/%s", algorithm.DatasetsDir, dataset.Filename))
```
Comment (WashingtonKK) [inline agent/service.go:197]: Handle an instance where the filename is empty so that you do not have a dataset without a name.

Comment (WashingtonKK) [reply agent/service.go:197]: i.e., if dataset is empty, save the dataset as dataset-1 and so on.

Comment (WashingtonKK) [reply agent/service.go:197]: Since file is uploaded from cli, the filename can not be empty so this approach is ok.

**Review by WashingtonKK — APPROVED** [review]

**Review by drasko — COMMENTED** [review]
LGTM

**Pull request merged**